### PR TITLE
Add Otter Browser

### PR DESCRIPTION
**Pull request by jurf**
[Otter Browser](https://otter-browser.org/) aims to recreate the best aspects of the classic Opera (12.x) UI using Qt5.

Easily the easiest Flatpak I’ve ever created. Kudos, @Emdek!

### CODE REVIEW

**Comment by jurf** [comment]
bot, build

**Comment by jurf** [comment]
bot, build org.otter_browser.OtterBrowser

File: org.otter_browser.OtterBrowser.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "app-id": "org.otter_browser.OtterBrowser",
```
Comment (jurf) [inline org.otter_browser.OtterBrowser.json]: Are we still replacing dashes with underscores in IDs?

File: org.otter_browser.OtterBrowser.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "--socket=pulseaudio"
+  ],
+  "build-options": {
+    "cflags": "-O2 -g",
+    "cxxflags": "-O2 -g"
+  },
+  "modules": [
+    {
+      "name": "otter-browser",
+      "buildsystem": "cmake",
```
Comment (barthalion) [inline org.otter_browser.OtterBrowser.json]: `cmake-ninja`

File: org.otter_browser.OtterBrowser.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  "base-version": "stable",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.9",
+  "sdk": "org.kde.Sdk",
+  "command": "otter-browser",
+  "rename-desktop-file": "otter-browser.desktop",
+  "rename-appdata-file": "otter-browser.appdata.xml",
+  "rename-icon": "otter-browser",
+  "finish-args": [
+    "--device=dri",
```
Comment (barthalion) [inline org.otter_browser.OtterBrowser.json]: No `xdg-download`?

File: org.otter_browser.OtterBrowser.json (hunk 1 above)
Comment (barthalion) [reply org.otter_browser.OtterBrowser.json]: Yes.

File: org.otter_browser.OtterBrowser.json (hunk 2 above)
Comment (jurf) [reply org.otter_browser.OtterBrowser.json]: Isn’t that experimental?

File: org.otter_browser.OtterBrowser.json (hunk 3 above)
Comment (jurf) [reply org.otter_browser.OtterBrowser.json]: Doesn’t help, still downloads things to /tmp. If downloading via ’Save as…’ then it produces an invalid file. Isn’t there a way to disable portals altogether?

**Comment by jurf** [comment]
No, it hasn’t been discussed before, I just went and done it :P

File: org.otter_browser.OtterBrowser.json (hunk 3 above)
Comment (TingPing) [reply org.otter_browser.OtterBrowser.json]: > Isn’t there a way to disable portals altogether? 

That is moving in the wrong direction, portals should be used or fixed if there are problems. Disabling them would just be a bad workaround.

File: org.otter_browser.OtterBrowser.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    "--share=network",
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--filesystem=xdg-download",
+    "--env=QTWEBENGINE_DISABLE_SANDBOX=1"
```
Comment (TingPing) [inline org.otter_browser.OtterBrowser.json]: So does it use webengine or webkit?

Comment (jurf) [reply org.otter_browser.OtterBrowser.json]: Can do both, you can choose, but WebEngine is more experimental.

**Comment by jurf** [comment]
bot, build org.otter_browser.OtterBrowser

File: org.otter_browser.OtterBrowser.json (hunk 4 above)
Comment (barthalion) [reply org.otter_browser.OtterBrowser.json]: Disabling sandboxing sounds worse than not having qtwebengine enabled at all.

File: org.otter_browser.OtterBrowser.json (hunk 3 above)
Comment (barthalion) [reply org.otter_browser.OtterBrowser.json]: Or it should be patched to properly download files where they belong;

File: org.otter_browser.OtterBrowser.json (hunk 4 above)
Comment (jurf) [reply org.otter_browser.OtterBrowser.json]: Well it doesn’t run in Flatpak and we have Bubblewrap anyway, I don’t see a problem with this, I took this from existing Flathub manifests.

Comment (Frenzie) [reply org.otter_browser.OtterBrowser.json]: @barthalion Revived QtWebKit is quite capable. No QtWebEngine isn't bad at all really. ;-)

**Comment by TingPing** [comment]
This was discussed briefly on IRC and since Otter does not yet have a stable release we will wait to add this to Flathub. We also hope by time it hits "stable" that Webrender support will be considered stable also.

**Comment by jurf** [comment]
Stable now available: https://thedndsanctuary.eu/index.php?topic=3574.0

However, QtWebEngine is still considered experimental.

File: org.otter_browser.OtterBrowser.json (hunk 4 above)
Comment (jurf) [reply org.otter_browser.OtterBrowser.json]: I dunno, keep sandbox disabled or disable WebEngine?

**Comment by jurf** [comment]
bot, build org.otter_browser.OtterBrowser

**Comment by LeandroStanger** [comment]
bot, build org.otter_browser.OtterBrowser

**Pull request closed without merging**